src/main/zencode/stdlib/src/Strings.zs

```diff
@@ -25,101 +27,83 @@ public expand string {
-		for i in from .. length {
-			if this[i] == c
-				return i;
-		}
-		
-		return null;
-	}
+	public const indexOf(c as char, from as usize) as usize?;
 	
 	[Native("indexOfString")]
 	public const indexOf(s as string) as usize?;
 	
 	[Native("indexOfStringFrom")]
 	public const indexOf(s as string, from as usize) as usize?;
-	
+
 	[Native("lastIndexOf")]
-	public const lastIndexOf(c as char) as usize? {
-		var i = length;
-		while i > 0 {
-			i--;
-			if this[i] == c
-				return i;
-		}
-		
-		return null;
-	}
-	
+	// ToDo: This had a method body before but also a native binding, which to use?
+	public const lastIndexOf(c as char) as usize?;
+
 	[Native("lastIndexOfFrom")]
-	public const lastIndexOf(c as char, until as usize) as usize? {
-		var i = until;
-		while i > 0 {
-			i--;
-			if this[i] == c
-				return i;
-		}
-		
-		return null;
-	}
-	
+	// ToDo: This had a method body before but also a native binding, which to use?
+	public const lastIndexOf(c as char, until as usize) as usize?;
+
 	[Native("lastIndexOfString")]
 	public const lastIndexOf(s as string) as usize?;
-	
+
 	[Native("lastIndexOfStringFrom")]
 	public const lastIndexOf(s as string, until as usize) as usize?;
-	
-	[Native("split")]
-	public const split(delimiter as char) as string[] {
-		val result = new List<string>();
-		var start = 0 as usize;
-		for i in 0 .. this.length {
-			if this[i] == delimiter {
-				result.add(this[start .. i]);
-				start = i + 1;
-			}
-		}
-		result.add(this[start .. $]);
-		return result as string[];
-	}
-	
-	[Native("trim")]
-	public const trim() as string {
-		var from = 0 as usize;
-		while from < this.length && this[from] in [' ', '\t', '\r', '\n']
-			from++;
-		var to = this.length;
-		while to > 0 && this[to - 1] in [' ', '\t', '\r', '\n']
-			to--;
-		
-		return to < from ? "" : this[from .. to];
-	}
-	
+
+
+    // ToDo: cannot infer i + 1 without a cast as usize
+    // ToDo: does not know the $ symbol
+    // ToDo: cannot cast List<T> to T[]
+	//[Native("split")]
+	//public const split(delimiter as char) as string[] {
+	//	val result = new List<string>();
+	//	var start = 0 as usize;
+	//	for i in 0 .. this.length {
+	//		if this[i] == delimiter {
+	//			result.add(this[start .. i]);
+	//			start = i + 1;
+	//		}
+	//	}
+	//	result.add(this[start .. $]);
+	//	return result as string[];
+	//}
+
+    // ToDo: invalid postfix expression
+	//[Native("trim")]
+	//public const trim() as string {
+	//	var from = 0 as usize;
+	//	while from < this.length && this[from] in [' ', '\t', '\r', '\n']
+	//		from++;
+	//	var to = this.length;
+	//	while to > 0 && this[to - 1] in [' ', '\t', '\r', '\n']
+	//		to--;
+	//	return to < from ? "" : this[from .. to];
+	//}
+
 	[Native("startsWith")]
-	public const startsWith(head as string) as bool
-		=> length >= head.length && this[0 .. head.length] == head;
-	
+	// ToDo: This had a method body before but also a native binding, which to use?
+	public const startsWith(head as string) as bool;
```

Comment:
❓ Question: Can a `[Native]` method have a body?

Reply:
Yes it can.
If it also has a definition inside Java Code, then that should be used, otherwise it should default to the method body.